test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -94,12 +95,22 @@ def setup_sc(self, sched_id, partition, port):
         :param log_filter: log filter value for the scheduler
         :type log_filter: int
         """
-        a = {'partition': partition,
-             'sched_host': self.server.hostname,
-             'sched_port': port}
+        if sched_priv is not None:
+            a = {'partition': partition,
+                 'sched_host': self.server.hostname,
+                 'sched_port': port,
+                 'sched_priv': os.path.join(sched_priv, 'sched_priv_sc3'),
+                 'sched_log': os.path.join(sched_log, 'sched_logs_sc3')}
+        else:
```

Comment:
See my comment above, I think it's cleaner if you build the attribute list with just these 3 attributes and then add 'sched_priv' and 'sched_log' conditionally.